src/ts/components/subtitleoverlay.ts

```diff
@@ -470,6 +477,24 @@ class ActiveSubtitleManager {
     }
   }
 
+  /**
+   * Removes all active cues which don't enclose the given time
+   * @param time the time for which subtitles should remain
+   */
+  public clearInactiveCues(time: number): ActiveSubtitleCue[] {
+    const removedEvents: ActiveSubtitleCue[] = [];
+    Object.keys(this.activeSubtitleCueMap).forEach(key => {
+      const activeCues = this.activeSubtitleCueMap[key];
+      activeCues.forEach(element => {
+        if (time < element.event.start || time > element.event.end) {
+          this.popCueFromMap(element.event);
+          removedEvents.push(element);
+        }
+      });
+    });
+    return removedEvents;
+  }
```

Comment:
So we are preserving only the cue that lies at the given time. Can it happen that there are cues in the future (i.e. that lie after `time`) that should not be cleared? 🤔

Reply:
I can't think of a reason for there being active cues which should linger until they become active again. I would expect the player to emit `CueEnter` events when it reaches that range. Then we would create an active cue again. If we didn't clear it before, there would then be two cues maybe?

Reply:
Thanks 👍 , I thought so, but wanted to ask if I understood it correctly. 🙂